src/core/classes/contract-publisher.ts

```diff
@@ -284,9 +284,18 @@ export class ContractPublisher extends RPCConnectionHandler {
         }
       }
       if (p === "bytes32") {
-        return ethers.utils.formatBytes32String(
-          constructorParamValues[index].toString(),
-        );
+        if (!isNaN(constructorParamValues[index]) && constructorParamValues[index] >= 0) {
```

Comment:
`constructorParamValues[index]` will always be a string - this check won't work.

Howw about `ethers.utils.isHexString(constructorParamValues[index])` instead (which means you don't have to hexlify under the hood, just pad)

Reply:
`isNaN('0x1234')` returns `false`
`isNaN('1234')` returns `false`

Reply:
@bagelface we use typescript, and in typescript `isNan()` expects a number, so it throws a type error (won't compile)

Reply:
but in that case their contract param type won't be `bytes32` right? would probably be `uint256` or something for a number parameter.

Reply:
Lol seems a little counter-intuitive to only accept a number for a "is not a number" check. We can wrap `constructorParamValues[index]` as `Number(constructorParamValues[index])` if you're ok with my previous comment about "1234" being treated like an integer